src/hivpy/simulation.py

```diff
@@ -82,38 +79,28 @@ def update_summary_stats(self, date, pop_data):
         self._update_deaths(pop_data)
         self.step += 1
 
-    def write_output(self):
+    def write_output(self, output_path):
         df = pd.DataFrame(self.output_stats)
-        df.to_csv(self.file_path, mode='w')
+        df.to_csv(output_path, mode='w')
 
 
 class SimulationHandler:
-    """A class for handling executing a simulation and accessing results."""
+    """A class for handling executing a simulation and outputting results."""
     simulation_config: SimulationConfig
     population: Population
-    results: pd.DataFrame
 
     def __init__(self, simulation_config):
         self.simulation_config = simulation_config
-        self.results = None
         self._initialise_population()
         self.output = SimulationOutput(self.simulation_config)
+        self.output_path = simulation_config.output_dir / (
+            "simulation_output." + str(datetime.now().strftime("%Y%m%d-%H%M%S")) + ".csv")
```

Comment:
Unrelated to these changes, but it might be better to have the filename as `simulation_output_{date}.csv` instead, to avoid the extra `.` which can be confusing/mess up how the name is displayed?

Reply:
It shouldn't cause any issues for the name being displayed (it doesn't on my Windows machine at least) but I don't mind changing the style (in which case we should change the style of the log output files as well which are done similarly).